### ISSUE TITLE
ProjectCodes should accommodate code look-up as well as set-based value constraints.

### ISSUE THREAD

**Issue by benjamin-heasly** [benjamin-heasly/openhds-rest#97]
The existing openhds-server has two kinds of "code" configuration in two files:
- `codes.properties` enables simple name -> value lookup
- `value-constraint.xml` enables constraints of the form “is value x a member of set Y”?

The ProjectCodes model and service should be extended accommodate both operations.

The ProjectCodes model currently has
- codeName
- codeValue

It should add
- codeGroup
- description

This model will accommodate all the data found in `codes.properties` and `value-constraint.xml`.

The ProjectCodes service should then expose operations of the form
- lookup (codeName) -> value
- predicate (codeGroup, value x) -> is x one of the codeValues in codeGroup?

The lookup assumes codeNames are still unique -- codeGroup is metadata, not a namespace.

So in summary, adding codeGroups to the model and service operations will make ProjectCodes more expressive and useful.

Adding a description to each code will just be handy.  It's a good idea from the existing openhds-server.

**Comment by benjamin-heasly** [comment]
f3d6c5db633c255db0067bff20386530dace536a contains this change.
